src/Autocomplete.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [noSuggestions, setNoSuggestions] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (inputValue.trim() === '') {
+        setSuggestions(people);
+        setNoSuggestions(false);
+
+        return;
+      }
+
+      const filteredPeople = people.filter(person =>
+        person.name.toLowerCase().includes(inputValue.toLowerCase()),
+      );
+
+      setSuggestions(filteredPeople);
+      setNoSuggestions(filteredPeople.length === 0);
+    }, delay);
+
+    return () => clearTimeout(timeoutId);
+  }, [inputValue, people, delay]);
+
+  const handleSelect = (person: Person) => {
+    setInputValue(person.name);
+    setSelectedPerson(person);
+    setSuggestions([]);
+    onSelected(person);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
It's recommended to follow the naming conventions for methods. The method name 'handleInputChange' is fine, but it's better to specify what is changing. A more descriptive name could be 'handleInputValueChange'.